cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1507,11 +1518,46 @@ protected String getServiceAccountName() {
         return varList;
     }
 
+    /**
+     * Validates provided OAuth configuration. Throws InvalidResourceException when OAuth configuration contains forbidden combinations.
+     *
+     * @param oAuth     OAuth type authentication object
+     */
+    @SuppressWarnings("checkstyle:BooleanExpressionComplexity")
+    private static void validateOauth(KafkaListenerAuthenticationOAuth oAuth) {
```

Comment:
Given that this method can be called up to 3 times for a give CR it would be helpful to the user if the error message indicated which listener's auth object has the problem.